src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -658,6 +660,22 @@ public bool ExtractNativeLibraries ()
 			return true;
 		}
 
+		/// <summary>
+		/// Returns true if an element has the @android:directBootAware attribute and its 'true'
+		/// </summary>
+		public bool DirectBootAware ()
+		{
+			var processAttrName = androidNs.GetName ("directBootAware");
+			foreach (XElement el in app.Elements ()) {
```

Comment:
Does this only need to look at certain types of elements?
```xml
<activity|provider|receiver|service ...  
     android:directBootAware=”true”>
```

Or does it matter?

Reply:
I spoke with jonp about that. I think we should pick up any directBootAware. That means it will be forward compatible if google decide to add a new one. That said I need to also check the app element itself as well. I'll fix that up today.